lib/puppet/provider/zone/solaris.rb

```diff
@@ -234,10 +234,10 @@ def getconfig
           end
           current[$1.intern] = $2
         else
-          err "Ignoring '#{line}'"
+          Puppet.err "Ignoring '#{line}'"
```

Comment:
The real root cause is that puppet assumes if the resource responds to `path` then it must respond to `tags`, `file`, etc. When that issue is fixed, then this change will cause us to lose context information about the resource that failed, e.g.

```
Error: /Stage[main]/Main/Zone[test_zone]: Ignoring ...
```

How about we fix the root cause instead, so we don't lose the context and ensure we work with custom providers that also have a path, e.g. registry_key. Something like?

```
diff --git a/lib/puppet/util/log.rb b/lib/puppet/util/log.rb
index 08a90b1..8909ab3 100644
--- a/lib/puppet/util/log.rb
+++ b/lib/puppet/util/log.rb
@@ -338,9 +338,9 @@ class Puppet::Util::Log
   def source=(source)
     if source.respond_to?(:path)
       @source = source.path
-      source.tags.each { |t| tag(t) }
-      self.file = source.file
-      self.line = source.line
+      source.tags.each { |t| tag(t) } if source.respond_to?(:tags)
+      self.file = source.file if source.respond_to?(:file)
+      self.line = source.line if source.respond_to?(:line)
     else
       @source = source.to_s
     end
```

cc'ing @hlindberg and @thallgren who were the last to work on logging messages.